### ISSUE TITLE
Pre-Release Version 2025.1.101 produces false error

### ISSUE THREAD

**Issue by rafrafek** [microsoft/pyright#9778]
## Environment data

-   Pylance version: Pre-Release 2025.1.101
-   OS and version: MacOS Sequoia 15.2 (Apple M1 Pro)
-   Python version (& distribution if applicable, e.g. Anaconda): Python 3.12.8 (Poetry 2.0.1)

## Code Snippet

```python
from aws_cdk import Stack, aws_ec2
from constructs import Construct


class DemoStack(Stack):
    def __init__(
        self,
        scope: Construct,
        construct_id: str,
        vpc_core_id: str,
    ) -> None:
        super().__init__(scope, construct_id)

        core_vpc = aws_ec2.Vpc.from_lookup(self, "LambdaVpc", vpc_id=vpc_core_id)  # Here!

        print(core_vpc)
```

## Repro Steps

1. Clone https://github.com/rafrafek/pylance-demo and open `pylance_demo/__init__.py`, or use the script above
2. Install all the requirements, preferably using Poetry 2.0.1, or just install the latest `aws-cdk-lib`
3. Switch between Pylance extension versions: Release Version 2024.12.1 and Pre-Release Version 2025.1.101 and observe an error appearing on the Pre-Release Version 2025.1.101 on line 14.

## Expected behavior

No error on both versions of Pylance.

## Actual behavior

Pre-Release Version 2025.1.101 produces an error on line 14:

```
pylance_demo/__init__.py:14

Argument missing for parameter "id".
```

## Logs

```
Please let me know if the error is not reproducible, I will attach my logs then.
```

**Comment by erictraut** [comment]
This issue is more complex than I originally thought. The core problem is not what I initially surmised above. I was correct that there are some code paths in pyright's binder logic that don't handle nonstandard ways of specifying a `classmethod`. I've enhanced these code paths to handle `builtins.classmethod`, but the problem still persists.

The root cause is that the source file in question is very large and complicated. It's over 110K lines, and the code flow graph for the global (module-level) scope has so many nodes that pyright decides that it's not feasible to perform code flow analysis on it. This internal limit prevents pyright from hanging when the type analyzer is presented with very complex code.

The reason you're seeing a difference in behavior is related to a bug fix that I introduced recently (it hasn't yet appeared in any published version of pyright). When the analyzer is asked to perform code flow analysis on a symbol to determine its locally-narrowed type _and_ it determines that the code flow graph exceeds the complexity threshold, it now returns a type of `Unknown`. The old behavior led to confusing false positive errors in some cases. The new behavior (of returning `Unknown` in this case) has an effect that I hadn't anticipated when I introduced the bug fix. In this particular case, the threshold is exceeded when pyright attempts to evaluate the `builtins` symbol. Since its type is now evaluated as `Unknown`, the expression `builtins.classmethod` is also evaluated as `Unknown`, and the decorator has no effect.

The real underlying problem here is that the top-level module for this library is just too large for a static analyzer to handle well. It should be broken up into submodules. As you noted above, opening the file in pylance causes it to consume all available heap space and crash. 

I've tweaked the internal threshold slightly to addresses the immediate issue that you reported. This will appear in the next release of pyright. However, this isn't a permanent fix. As this file inevitably grows larger, there will soon come a time when it won't work at all with pyright or pylance.

If this is a library that's important to you, I encourage you to file an issue with its maintainers. Since the file appears to be machine-generated, I would think that it would be relatively little work for them to generate multiple submodules that are imported by a top-level module.

**Comment by erictraut** [comment]
This is addressed in pyright 1.1.393.

**Comment by rafrafek** [comment]
Thank you for addressing the issue! 🙏
